pymongo/synchronous/client_session.py

```diff
@@ -598,7 +600,10 @@ def _inherit_option(self, name: str, val: _T) -> _T:
 
     def with_transaction(
         self,
-        callback: Callable[[ClientSession], _T],
+        callback: Union[
+            Callable[[ClientSession], _T],
+            Callable[[ClientSession], _T],
+        ],
```

Comment:
It looks like the type definition is the same twice here. Is that needed?

Reply:
No, this is an artifact of supporting both kinds of callbacks in the async API. Will be resolved in the separate PR for `test_transaction`